mergify_engine/github_events.py

```diff
@@ -280,7 +290,30 @@ async def push_to_worker(
         elif event["repository"]["archived"]:  # pragma: no cover
             ignore_reason = "repository archived"
 
-        await context.Repository.clear_config_file_cache(redis_links.cache, repo_id)
+        if f"refs/heads/{event['repository']['default_branch']}" == event["ref"]:
+
+            # NOTE(sileht): commits contains the list of commits returned by compare API
+            # that by default returns only 250 commits
+            # https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push
+            # https://docs.github.com/en/rest/commits/commits#compare-two-commits
+            if event["forced"] or len(event["commits"]) > 250:
+                mergify_configuration_changed = True
+            else:
+                mergify_configuration_changed = False
+
+                mergify_config_filenames = set(constants.MERGIFY_CONFIG_FILENAMES)
+                for commit in event["commits"]:
+                    if (
+                        set(commit["added"] + commit["removed"] + commit["modified"])
+                        & mergify_config_filenames
+                    ):
```

Comment:
```suggestion
                    if (
                        set(commit["added"] & mergify_config_filenames or commit["removed"] & mergify_config_filenames or set(commit["modified"]) & mergify_config_filenames
                    ):
```
if you want to optimize further

Reply:
even better